src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,120 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROW_FACTOR = 1.5;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity();
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForIndexAdd(index);
+        ensureCapacity();
+        for (int i = size; i > index; i--) {
+            elementData[i] = elementData[i - 1];
+        }
+        elementData[index] = value;
+        size++;
 
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list == null) {
+            throw new NullPointerException("The list cannot be null!");
+        }
+        if (list.isEmpty()) {
+            throw new IllegalArgumentException("The list cannot be empty!");
+        }
```

Comment:
You are throwing an IllegalArgumentException if the list is empty. However, adding an empty list to the ArrayList should be a valid operation and should not throw an exception. Consider removing this check.